server/src/main/java/org/opensearch/index/shard/StoreRecovery.java

```diff
@@ -463,9 +467,32 @@ private void recoverFromRemoteStore(IndexShard indexShard) throws IndexShardReco
             for (String file : storeDirectory.listAll()) {
                 storeDirectory.deleteFile(file);
             }
+            String segmentInfosSnapshotFilename = null;
             for (String file : remoteDirectory.listAll()) {
                 storeDirectory.copyFrom(remoteDirectory, file, file, IOContext.DEFAULT);
+                if (file.startsWith(SEGMENT_INFO_SNAPSHOT_FILENAME_PREFIX)) {
+                    segmentInfosSnapshotFilename = file;
+                }
             }
+
+            if (segmentInfosSnapshotFilename != null) {
+                try (
+                    ChecksumIndexInput indexInput = new BufferedChecksumIndexInput(
+                        storeDirectory.openInput(segmentInfosSnapshotFilename, IOContext.DEFAULT)
+                    )
+                ) {
+                    SegmentInfos infosSnapshot = SegmentInfos.readCommit(
+                        store.directory(),
+                        indexInput,
+                        Long.parseLong(segmentInfosSnapshotFilename.split("__")[1])
+                    );
+                    long processedLocalCheckpoint = Long.parseLong(infosSnapshot.getUserData().get(LOCAL_CHECKPOINT_KEY));
+                    store.commitSegmentInfos(infosSnapshot, processedLocalCheckpoint, processedLocalCheckpoint);
+                } catch (IOException e) {
+                    logger.info("Exception while reading {}, falling back to commit level restore", segmentInfosSnapshotFilename);
+                }
+            }
```

Comment:
Need better logging exception handling here, we need to know when could this happen and avoid silent pass through

Reply:
This was called out earlier as well: https://github.com/opensearch-project/OpenSearch/pull/5253#discussion_r1029306959. I agree that `info` log level does not provide much context of the scenario. I will get it changed to `warn`. 

Currently, this is a defensive mechanism which allows the restore to continue even if we failed to restore from `SegmentInfosSnapshot` file. Instead of silently moving ahead with commit level durability, we can take it as an input to the API or in terms of config.